src/seq/index.rs

```diff
@@ -249,6 +252,114 @@ where R: Rng + ?Sized {
     }
 }
 
+/// Randomly sample exactly `amount` distinct indices from `0..length`, and
+/// return them in an arbitrary order (there is no guarantee of shuffling or
+/// ordering). The weights are to be provided by the input function `weights`,
+/// which will be called once for each index.
+///
+/// This method is used internally by the slice sampling methods, but it can
+/// sometimes be useful to have the indices themselves so this is provided as
+/// an alternative.
+///
+/// This implementation uses `O(length)` space and `O(length)` time if the
+/// "partition_at_index" feature is enabled, or `O(length)` space and
+/// `O(length * log amount)` time otherwise.
+///
+/// Panics if `amount > length`.
+pub fn sample_weighted<R, F, X>(
+    rng: &mut R, length: usize, weight: F, amount: usize,
+) -> Result<IndexVecIntoIter, WeightedError>
```

Comment:
Why `IndexVecIntoIter` instead of `IndexVec`?

Reply:
Hmm, I don't think there's a good reason for that here.